com.unity.multiplayer.mlapi/Tests/Runtime/Metrics/NetworkObjects/NetworkObjectMetricsTests.cs

```diff
@@ -0,0 +1,161 @@
+            Assert.AreEqual(m_Server.LocalClientId, objectSpawned.Connection.Id);
+            Assert.AreEqual(networkObject.NetworkObjectId, objectSpawned.NetworkId.NetworkId);
+            // TODO: this should not be the name of the network object
+            Assert.AreEqual("Player(Clone)", objectSpawned.NetworkId.Name); // What?
+        }
+
+        [UnityTest]
+        public IEnumerator TrackNetworkObjectDestroySentMetric()
+        {
+            var objectName = "TestNetworkObjectToDestroy";
+            var gameObject = new GameObject(objectName);
+            var networkObject = gameObject.AddComponent<NetworkObject>();
+            networkObject.NetworkManagerOwner = m_Server;
+            networkObject.Spawn();
+
+            var waitForMetricEvent = new WaitForMetricValues<ObjectDestroyedEvent>(m_ServerMetrics.Dispatcher, MetricNames.ObjectDestroyedSent);
+            // TODO: is there a better way of waiting here?
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            m_Server.SpawnManager.OnDestroyObject(networkObject.NetworkObjectId, true);
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            var objectDestroyedSentMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            // As there's a client and server, this event is emitted twice.
+            Assert.AreEqual(2, objectDestroyedSentMetricValues.Count);
+
+            var objectDestroyed = objectDestroyedSentMetricValues.Last();
+            Assert.AreEqual(m_Client.LocalClientId, objectDestroyed.Connection.Id);
+            Assert.AreEqual(objectName, objectDestroyed.NetworkId.Name);
+        }
+
+        [UnityTest]
+        public IEnumerator TrackNetworkObjectDestroyReceivedMetric()
+        {
+            var objectName = "TestNetworkObjectToDestroy";
+            var gameObject = new GameObject(objectName);
+            var networkObject = gameObject.AddComponent<NetworkObject>();
+            networkObject.NetworkManagerOwner = m_Server;
+
+            networkObject.Spawn();
+            var waitForMetricEvent = new WaitForMetricValues<ObjectDestroyedEvent>(m_ClientMetrics.Dispatcher, MetricNames.ObjectDestroyedReceived);
+
+            yield return waitForMetricEvent.WaitForAFewFrames();
+
+            m_Server.SpawnManager.OnDestroyObject(networkObject.NetworkObjectId, true);
+            yield return waitForMetricEvent.Wait(60);
+
+            var objectDestroyedReceivedMetricValues = waitForMetricEvent.EnsureMetricValuesHaveBeenFound();
+            Assert.AreEqual(1, objectDestroyedReceivedMetricValues.Count);
+
+            var objectDestroyed = objectDestroyedReceivedMetricValues.First();
+            Assert.AreEqual(m_Server.LocalClientId, objectDestroyed.Connection.Id);
+            Assert.AreEqual(networkObject.NetworkObjectId, objectDestroyed.NetworkId.NetworkId);
+            // Assert.AreEqual(, objectDestroyed.NetworkId.Name); // What?
```

Comment:
Do we want to log a bug for this?

Reply:
yes I will see what the test does as iirc it used the wrong name/"Player (clone)" regardless, but just want to double check that especially now we've merged with develop branch a few times